src/apps/VirtualApp/VirtualClient.js

```diff
@@ -126,10 +126,13 @@ class VirtualClient extends Component {
     this.state.virtualStreamingJanus.destroy();
   }
 
-  checkPermission = (user) => {
+  checkPermission = (user, access_token) => {
+    api.setAccessToken(access_token);
```

Comment:
The listeners should only happen in the if (firstTime)
Otherwise you will register to the event several times.

Reply:
Keep api.setAccessToken(access_token); here.

Move
client.events.addUserLoaded((user) => api.setAccessToken(user.access_token));
    client.events.addUserUnloaded(() => api.setAccessToken(null));

To where it was.

Reply:
checkPermission called once. so this code is correct.